src/main/java/com/google/gwt/dom/client/DOMImpl.java

```diff
@@ -19,15 +19,108 @@
 
 import com.google.common.base.MoreObjects;
 import com.google.gwt.core.client.JsArray;
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.ChangeEvent;
 import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.FocusEvent;
+import com.google.gwt.event.dom.client.DomEvent.Type;
+import com.google.gwt.user.client.DOM;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.EventListener;
 
 public class DOMImpl {
 	
 	static final DOMImpl impl = new DOMImpl();
 	
-	public void buttonClick(ButtonElement button) {
-		button.fireEvent(ClickEvent.getType());
-	}
+	// Element In Focus (some browsers actually have working implementation document.activeElement)
+	private Element activeElement;
+	
+    private void switchFocusTo(Element element) {
+        if (activeElement == element) {
+            return;
+        } else {
+            if (activeElement != null && isAtached(activeElement)) {
+                if (activeElement instanceof InputElement) {
+                    fireValueChangeIfAny((InputElement)activeElement);
+                }
+                activeElement.fireEvent(BlurEvent.getType());
+            }
+            element.fireEvent(FocusEvent.getType());
+            activeElement = element;
+        }
+    }
+	
+    private boolean isAtached(Element elem) {
```

Comment:
I think there's a slight but important difference: fireEvent on element is just GWT-side pair of Element.dispatchEvent, and as __listener and __eventbits are also naturally there on the Element (as they are there internally in a real browser too), the fireEvent method is also naturally there. So it's rather a part of the 'browser-simulation' than a test utility.